### PR TITLE
Updating config to use custom image and adding image for e2e tests

### PR DESCRIPTION
**Pull request by mohammedzee1000**

### CODE REVIEW

File: ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml [hunk 1]
```diff
@@ -8,32 +8,40 @@ resources:
...
-    #      - destination_dir: .
-    #        source_path: /usr/bin/oc
-    #  to: your-output-image
+build_root:
+  project_image:
+    dockerfile_path: openshift-ci/build-root/Dockerfile
+
+binary_build_commands: make bin
+
+images:
```
Comment (stevekuznetsov) [inline ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml]: Do you need this?

Comment (mohammedzee1000) [reply ci-operator/config/redhat-developer/odo/redhat-developer-odo-master.yaml]: No, removing the same

**Comment by mohammedzee1000** [comment]
@stevekuznetsov is the installer config on OpenShift CI generated in the same way the normal installer one is as in is it placed in `$(pwd)/auth/kubeconfig` and `$(pwd)/auth/kubeadmin-password

**Comment by amitkrout** [comment]
https://github.com/redhat-developer/odo/issues/1431

**Comment by mohammedzee1000** [comment]
/hold cancel
PTAl @stevekuznetsov

File: ci-operator/jobs/redhat-developer/odo/redhat-developer-odo-master-presubmits.yaml [hunk 2]
```diff
@@ -1,5 +1,127 @@
...
+        - name: JOB_NAME_SAFE
+          value: e2e
+        - name: TEST_COMMAND
+          value: |
+            make configure-installer-tests-cluster
+            export PATH="$PATH:$(pwd)"
+            export CUSTOM_HOMEDIR="/tmp/artifacts"
+            make test-login-e2e
+            make test-json-format-output
+            make test-java-e2e
```
Comment (amitkrout) [inline ci-operator/jobs/redhat-developer/odo/redhat-developer-odo-master-presubmits.yaml]: Don't use ```test-java-e2e``` as i am seeing some failure while trying it locally. I am kind of sure that same failure will be observed on prow too. 

So remove ```make test-java-e2e``` as of now. Will keep adding one by one after fixing the issue from test side.

Comment (mohammedzee1000) [reply ci-operator/jobs/redhat-developer/odo/redhat-developer-odo-master-presubmits.yaml]: Ok

Comment (amitkrout) [reply ci-operator/jobs/redhat-developer/odo/redhat-developer-odo-master-presubmits.yaml]: ```make test-login-e2e``` and ```make test-json-format-output``` looks good to me

**Pull request merged**